### PR TITLE
Personal AIs can now access the crew manifest

### PR DESCRIPTION
**Pull request by JustinTrotter**
## About the PR

Set up a new action for pAIs to view the crew manifest. Should only apply to the station that they are currently on and only if the CCvars are set to allow unsecure crew manifest viewing.

**Media**

screenshot_pai_crew_manifest

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Personal AIs can now access the crew manifest

### CODE REVIEW

**Comment by JustinTrotter** [comment]
If/when this gets merged. I'll add something similar for ghosts.

**Comment by AJCM-git** [comment]
Cant you use a `IntrinsicUiComponent`?

File: Content.Server/CrewManifest/CrewManifestSystem.cs [hunk 1]
```diff
@@ -135,6 +137,23 @@ private void OpenEuiFromBui(EntityUid uid, CrewManifestViewerComponent component
         OpenEui(owningStation.Value, sessionCast, uid);
     }
 
+    private void OnViewManifestAction(EntityUid uid, CrewManifestViewerComponent component, ViewManifestActionEvent msg)
+    {
+        _entityManager.TryGetComponent(uid, out ActorComponent? actor);
+        var owningStation = _stationSystem.GetOwningStation(uid);
+        if (owningStation == null || actor == null || actor.PlayerSession is not IPlayerSession sessionCast)
```
Comment (metalgearsloth) [inline Content.Server/CrewManifest/CrewManifestSystem.cs:144]: You shouldn't copy-paste code between methods like this as any bugfixes need to be copy-pasted assuming whoever fixes them sees them both. They should both be calling the same method.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by AJCM-git** [comment]
@JustinTrotter you coming back to this?

**Pull request closed without merging**